test/TesterInternal/StorageTests/SQLAdapter/temp/TestEnvironmentInformation.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace UnitTests.StorageTests.SQLAdapter.temp
+{
+    /// <summary>
+    /// Enumeration of the supported test environment configurations.
+    /// </summary>
+    public enum SupportedTestConfigurations
+    {
+        None                        = 0,
+        Development                 = 1,
+        Jenkins                     = 2,
+        VisualStudioTeamServices    = 3
+    }
```

Comment:
This is not used to select a combination of tests. This enumerates the platforms where tests are assumed to be run because the enumeration value is easier to pass around. It is used like [here](https://github.com/dotnet/orleans/pull/1682/files#diff-4b8eb4bc30b3f20f1d850ea8cbdf9fc5R57) in the switch. The functionality that detects the correction enumeration value is [harcoded](https://github.com/dotnet/orleans/pull/1682/files#diff-a0630db3acf1d2e9ce2c4751bb908aecR29) currently, because I didn't want to go all the way before asking feedback.

In reality the implementation could check, for instance, if [JENKINS_HOME](http://www.tutorialspoint.com/jenkins/jenkins_configuration.htm) is set (and that it is not in VSTS) and so determine the test suite is being run in GitHub Jenkins and check invariants hold accordingly -- like that what tests to run and what to skip (i.e. MySQL installed, run them, skip Azure Emulator ones, as for an example, other combinations would be OK too). For VSTS this could check path variables too and then when checking environment invariants hold, set a hosted LocalDb build -- which is [a bit involved](https://blogs.msdn.microsoft.com/ssdt/2014/07/24/sql-server-database-projects-and-team-foundation-build/). Or that in VSTS one could get connection strings from Key Vault, or prepare for more and different tests to be run, in the same suite.

I know this is asking a bit of a imagination here as this is not ready. I'm seeking for opinions if this would like an OK direction to take without actually using more time and going all the way down this path. What I think is that it would be beneficial to be able to run a broad set of tests in all the environments and it looks to me it makes sense to make the invariants checks, or "prepare the system" explicit as-per platform. It looks also that doing it this way allows one to run tests in parallel.

Some of this could be done to some extent in test class constructors and inheritance, but it would kind of mean tests are setup or that it is not that easy to move invariant checks (or otherwise check and prepare) in different [levels of sharing](https://xunit.github.io/docs/shared-context.html).